src/plugins/ciu/ciuPlugin.js

```diff
@@ -0,0 +1,117 @@
+const fs = require('fs');
+const { messageToFactoid } = require('../factoids/factoidsPlugin');
+
+const AGENTS_TO_SHOW = [
+  'ie',
+  'edge',
+  'firefox',
+  'chrome',
+  'opera',
+  'safari',
+  'ios_saf',
+  'android',
+];
+
+function getVersions(agents, feature) {
+  return AGENTS_TO_SHOW.flatMap((agent) => {
+    // agents versions strings are sorted
+    const versions = agents[agent].versions.filter(Boolean);
+    const versionFull = versions.find(
+      (v) => v && !/[nupac]/g.test(feature.stats[agent][v]),
+    );
+
+    if (versionFull) return [`${agent}${versionFull}`];
+
+    const versionPartial = versions.find(
+      (v) => v && !/[nu]/g.test(feature.stats[agent][v]),
+    );
+
+    if (versionPartial) return [`${agent}${versionPartial}~`];
+
+    return [];
+  });
+}
+
+function perms(xs) {
+  if (xs.length === 1) return [xs];
+  return xs.flatMap((x, i) =>
+    perms([...xs.slice(0, i), ...xs.slice(i + 1)]).flatMap((vs) => [vs, [x, ...vs]]),
+  );
+}
+
+module.exports = async function ciuPlugin(msg) {
+  if (!msg.command) return;
+
+  const [cmd, ...search] = msg.command.command.split(' ');
+  if (cmd !== 'ciu' || search.length === 0) {
+    return;
+  }
+
+  const factoid = await messageToFactoid(msg);
+  if (factoid) {
+    return;
+  }
+
+  msg.handling();
+
+  // const db = require('caniuse-db/data.json'); // should we use the non-blocking way below? the json is quite large (2.7M)
+  const db = JSON.parse(
+    await fs.promises
+      .readFile(`${__dirname}/../../../node_modules/caniuse-db/data.json`, 'utf8')
```

Comment:
```suggestion
      .readFile(require.resolve('caniuse-db/data.json'), 'utf8')
```

This is fine, but the downside is that the require cache won't hold it for us.